app.py

```diff
@@ -53,33 +59,44 @@ def destroy_app(self):
     def run_app(self):
         """
         Callback from when the menu is clicked.
+
+        The default callback will first query the frame range from shotgun and validate the data.
+        If there is missing Shotgun data it will popup a QMessageBox dialog alerting the user.
+
+        Assuming all data exists in shotgun, it will set the frame range with the newly
+            queried data and popup a QMessageBox with results.
+
         """
 
         (new_in, new_out) = self.get_frame_range_from_shotgun()
-        (current_in, current_out) = self.get_current_frame_range(self.engine.name)
+        (current_in, current_out) = self.get_current_frame_range()
 
         if new_in is None or new_out is None:
             message =  "Shotgun has not yet been populated with \n"
             message += "in and out frame data for this Shot."
             QtGui.QMessageBox.information(None, "No data in Shotgun!", message)
             return
-            
+
         # now update the frame range.
         # because the frame range is often set in multiple places (e.g render range,
         # current range, anim range etc), we go ahead an update every time, even if
         # the values in Shotgun are the same as the values reported via get_current_frame_range()
-        self.set_frame_range(self.engine.name, new_in, new_out)
-        
-        message =  "Your scene has been updated with the \n"
-        message += "latest frame ranges from shotgun.\n\n"
-        message += "Previous start frame: %s\n" % current_in
-        message += "New start frame: %s\n\n" % new_in
-        message += "Previous end frame: %s\n" % current_out
-        message += "New end frame: %s\n\n" % new_out
-        
-        QtGui.QMessageBox.information(None, "Frame range updated!", message)
+        try:
```

Comment:
@willis102 I think perhaps the try should move to encompass all the run_app logic, ie move it up to line 57, as you would want the message box if any of these points failed. What do you think?